File: csharp/lib/Property/RoiProperty.cs
/*
 * SPDX-FileCopyrightText: 2020-2022 Sony Semiconductor Solutions Corporation
 *
 * SPDX-License-Identifier: Apache-2.0
 */

using MessagePack;

namespace SensCord
{
    /// <summary>
    /// Set the ROI information.
    /// </summary>
    [MessagePackObject]
    public class RoiProperty : IBaseProperty
    {
        /// <summary>
        /// Property key.
        /// </summary>
        public static readonly string ConstKey = "roi_property";

        /// <summary>
        /// Property key.
        /// </summary>
        [IgnoreMember]
        public string Key { get; } = ConstKey;

        /// <summary>
        /// The X coordinate of left top corner of ROI.
        /// </summary>
        [Key("left")]
        public int Left { get; set; }

        /// <summary>
        /// The Y coordinate of left top corner of ROI.
        /// </summary>
        [Key("top")]
        public int Top { get; set; }

        /// <summary>
        /// The width of the ROI.
        /// </summary>
        [Key("width")]
        public int Width { get; set; }

        /// <summary>
        /// The height of the ROI.
        /// </summary>
        [Key("height")]
        public int Height { get; set; }

        /// <summary>
        /// To string.
        /// </summary>
        /// <returns>A string that represents the current object.</returns>
        public override string ToString()
        {
            return string.Format(
                "{0}(left={1} top={2} width={3} height={4})",
                this.Key, this.Left, this.Top, this.Width, this.Height);
        }
    }
}

File: java/senscord/src/senscord_properties/RoiProperty.java
/*
 * SPDX-FileCopyrightText: 2020 Sony Semiconductor Solutions Corporation
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package senscord_properties;

import com.sun.jna.Structure;
import java.util.Arrays;
import java.util.List;

/**
 * Property for ROI.
 * 
 * @see "senscord::RoiProperty"
 */
public class RoiProperty extends Structure {
    public int left;
    public int top;
    public int width;
    public int height;

    public static class ByReference extends RoiProperty implements Structure.ByReference {
    }

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("left", "top", "width", "height");
    }

    /**
     * Get left.
     *
     * @return left
     */
    public int GetLeft() {
        return left;
    }

    /**
     * Get top.
     *
     * @return top
     */
    public int GetTop() {
        return top;
    }

    /**
     * Get width.
     *
     * @return width
     */
    public int GetWidth() {
        return width;
    }

    /**
     * Get Height.
     *
     * @return Height
     */
    public int GetHeight() {
        return height;
    }

    /**
     * Set left.
     *
     * @param left Left
     */
    public void SetLeft(int left) {
        this.left = left;
    }

    /**
     * Set top.
     *
     * @param top Top
     */
    public void SetTop(int top) {
        this.top = top;
    }

    /**
     * Set width.
     *
     * @param width Width
     */
    public void SetWidth(int width) {
        this.width = width;
    }

    /**
     * Set Height.
     *
     * @param height Height
     */
    public void SetHeight(int height) {
        this.height = height;
    }
}
